### ISSUE TITLE
Check if conversation ID is valid for all server endpoints

### ISSUE THREAD

**Issue by ricwo** [RasaHQ/rasa#7022]
**Description of Problem**:
In the HTTP server, we should check for endpoints that depend on `conversation_id` that this ID exists.

**Overview of the Solution**:
Create a decorator similar to `ensure_agent_is_loaded()` that uses the processor and checks if the conversation ID exists. If not, it returns an appropriate error code.

**Comment by aleronupe** [comment]
Hi @ricwo, @RomuloSouza and I got interested on this issue and we started to develop a solution to it. There are any other requirements to completely solve the problem than creating the decorator and raising the exception if the id doesn’t exist?

**Comment by ricwo** [comment]
cc @wochinge, would you mind taking this over? 🙏

**Comment by wochinge** [comment]
@aleronupe I don't think it necessarily needs to be a decorator. Not sure if it's possible, but it would also be great if we don't retrieve the tracker twice from the database.

And I don't think we need this warning for every endpoint. After having a quick look, I'd think that these would be candidates:
* `GET /conversations/<conversation_id:path>/story`
* maybe: `POST /conversations/<conversation_id:path>/execute`
* `POST /conversations/<conversation_id:path>/predict`

**Comment by aleronupe** [comment]
@wochinge We tried to develop a solution where we wouldn't have to retrieve the tracker twice from the database but, as each server endpoint uses a different processor function, we had to retrieve the tracker at the decorator in a more generic way.
